Shell/SymCounter.cpp

```diff
@@ -267,18 +275,25 @@ void SymCounter::count(Term* term, int polarity, int add)
     } else {
       //There should never be a non-shared sort
       int fun = term->functor();
+      ASS(!term->isSort());
       ASS_REP(_noOfFuns > fun, term->toString());
       _funs[fun].add(add);
 
-      NonVariableIterator nvi(term);
-      while (nvi.hasNext()) {
-        count(nvi.next().term(), 1, add);
+      for(TermList* arg=term->args(); arg->isNonEmpty(); arg=arg->next()) {
+        if(arg->isTerm()){
+          count(arg->term(), 1, add);
+        }
       }
     }
   } else {
     int fun = term->functor();
-    ASS_REP(_noOfFuns > fun, term->toString());
-    _funs[fun].add(add);
+    if(!term->isSort()){
```

Comment:
We never have predicates in term position...right?

Reply:
Correct. In FOOL and HOL, the predicates in term positions are treated as special terms, not predicates.